src/Tank.cpp

```diff
@@ -65,7 +65,21 @@ void Tank::update() {
 float Tank::get_tower_rotation() const { return mTower.get_rotation(); }
 
 void Tank::draw(sf::RenderWindow &window) {
-  update();
+  if (is_not_on_border()) {
+    update();
+  }
   mBody.draw(window, mPos.x, mPos.y);
   mTower.draw(window, mPos.x, mPos.y);
 }
+
+bool Tank::is_not_on_border() {
+  auto [x,y]=mPos;
+  float rotation=mBody.get_rotation();
+  float xcheck=mCurrentSpeed*(rotation-180);
+  float ycheck=mCurrentSpeed*(rotation-90)*(rotation-270);
+
+  if ((x<20 && xcheck>0) || (x>WIDTH-20 && xcheck<0) || (y<20 && ycheck>0) || (y>HEIGHT-20 && ycheck<0)) {
```

Comment:
I think I don't understand logic here. My initial idea of this issue was to:
- in method `update`, instead of changing `mPos`, create new variable, like `new_pos`
- if new `x` is inside board, update it
- if new `y` is inside board, update it

Right now I'm really confused about calculation done here. What is this xcheck & ycheck? Is it different way of calculating x & y? Maybe I complicated calculating position and it needs to be changed? For sure we shouldn't have two ways of calculating the position!

Reply:
The concept is that xcheck and ycheck verify the rotation and the speed of the tank, so if I'm moving to the left with +5 speed I can exit if my rotation is between 180 and 360, so if I subtract 180 and multiply the quantity is positive, the same woth the other situation.

Reply:
That's a little bit messy. What do you say about my proposition? Take a look, we have now: 
```cpp
  const auto rotation_degree = mBody.get_rotation() - 90;
  const auto rotation_radians = pi / 180.f * rotation_degree;
  mPos.x += mCurrentSpeed * std::cos(rotation_radians);
mPos.y += mCurrentSpeed * std::sin(rotation_radians);
``` - we are calculating new position. How about storing the position in new variable, like:
```cpp
  const auto rotation_degree = mBody.get_rotation() - 90;
  const auto rotation_radians = pi / 180.f * rotation_degree;
  const auto new_x += mPos.x + mCurrentSpeed * std::cos(rotation_radians);
  const auto new_y += mPos.y + mCurrentSpeed * std::sin(rotation_radians);
  if (is_inside_x(new_x)) {
    mPos.x = new_x;
  }
  if (is_inside_y(new_y)) {
    mPos.y = new_y;
  }
```
You will have to implement two methods: is_inside_x & is_inside_y (and fix any issues in above code :grin: ). 
Pros are that we are not doing new calculations & tank will be still moving along the edge. What do you think?